src/error.rs

```diff
@@ -19,11 +19,63 @@ use yaml_rust::scanner::{self, Marker, ScanError};
 use serde::{de, ser};
 
 use path::Path;
-
-/// This type represents all possible errors that can occur when serializing or
 /// deserializing YAML data.
 pub struct Error(Box<ErrorImpl>);
 
+#[derive(Clone, Debug)]
+pub struct ErrorPos {
+    pub message: String,
+    pub marker: Option<ErrorMarker>,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug, Eq)]
+pub struct ErrorMarker {
+    index: usize,
+    line: usize,
+    col: usize,
+}
+
+impl ErrorMarker {
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    pub fn line(&self) -> usize {
+        self.line
+    }
+
+    pub fn col(&self) -> usize {
+        self.col
+    }
+}
+
+impl<'a> From<&'a Marker> for ErrorMarker {
```

Comment:
This is still yaml-rust in the public API of serde-yaml. Imagine this code:

```toml
[dependencies]
yaml-rust = "0.3"
serde_yaml = "0.7"
```

```rust
use yaml_rust::scanner::Marker;
use serde_yaml::ErrorMarker;

fn x(marker: &Marker) -> ErrorMarker {
    From::from(marker)
}
```

This code would fail to compile after we bump yaml-rust version.